contracts/contracts/strategies/plume/RoosterAMOStrategy.sol

```diff
@@ -0,0 +1,1181 @@
+     */
+    function withdraw(
+        address _recipient,
+        address _asset,
+        uint256 _amount
+    ) external override onlyVault nonReentrant {
+        require(_asset == WETH, "Unsupported asset");
+        require(_amount > 0, "Must withdraw something");
+        require(_recipient == vaultAddress, "Only withdraw to vault allowed");
+
+        _ensureWETHBalance(_amount);
+
+        _withdraw(_recipient, _amount);
+    }
+
+    /**
+     * @notice Withdraw WETH and sends it to the Vault.
+     */
+    function withdrawAll() external override onlyVault nonReentrant {
+        if (tokenId != 0) {
+            _removeLiquidity(1e18);
+        }
+
+        uint256 _balance = IERC20(WETH).balanceOf(address(this));
+        if (_balance > 0) {
+            _withdraw(vaultAddress, _balance);
+        }
+    }
+
+    function _withdraw(address _recipient, uint256 _amount) internal {
+        IERC20(WETH).safeTransfer(_recipient, _amount);
+        emit Withdrawal(WETH, address(0), _amount);
+    }
+
+    /**
+     * @dev Retuns bool indicating whether asset is supported by strategy
+     * @param _asset Address of the asset
+     */
+    function supportsAsset(address _asset) public view override returns (bool) {
+        return _asset == WETH;
+    }
+
+    /**
+     * @dev Approve the spending amounts for the assets
+     */
+    function _approveTokenAmounts(
+        uint256 _wethAllowance,
+        uint256 _oethBAllowance
+    ) internal {
+        IERC20(WETH).approve(address(liquidityManager), _wethAllowance);
+        IERC20(OETHp).approve(address(liquidityManager), _oethBAllowance);
+    }
+
+    /***************************************
+              Liquidity management
+    ****************************************/
+    /**
+     * @dev Add liquidity into the pool in the pre-configured WETH to OETHp share ratios
+     * defined by the allowedPoolWethShareStart|End interval.
+     *
+     * Normally a PoolLens contract is used to prepare the parameters to add liquidity to the
+     * Rooster pools. It has some errors when doing those calculation and for that reason a
+     * much more accurate Quoter contract is used. This is possible due to our requirement of
+     * adding liquidity only to one tick - PoolLens supports adding liquidity into multiple ticks
+     * using different distribution ratios.
+     */
+    function _addLiquidity() internal {
+        uint256 _wethBalance = IERC20(WETH).balanceOf(address(this));
+        uint256 _oethBalance = IERC20(OETHp).balanceOf(address(this));
+        // don't deposit small liquidity amounts
+        if (_wethBalance <= MIN_BALANCE_THRESHOLD) {
+            return;
+        }
+
+        (
+            bytes memory packedSqrtPriceBreaks,
+            bytes[] memory packedArgs,
+            ,
+            uint256 OETHpRequired
+        ) = _getAddLiquidityParams(_wethBalance, 1e30);
+
+        if (OETHpRequired > _oethBalance) {
```

Comment:
Thought this code we use "oeth", "oethb", and "oethp" interchangeably. This gets to be a little confusing.

I think we shouldn't add the "oethp"'s. That's at least one less thing.

Event names should probably match the previous AMOs, so the backend team doesn't have to do extra work, and support multiple naming schemes.

Beyond that, my slight preference would be to shift the "oethb"'s back to "oeth", though up for second opinions there.